src/tree/updater_gpu_hist.cu

```diff
@@ -901,15 +951,21 @@ struct GPUHistMakerDevice {
       int left_child_nidx = tree[candidate.nid].LeftChild();
       int right_child_nidx = tree[candidate.nid].RightChild();
       // Only create child entries if needed
-      if (ExpandEntry::ChildIsValid(param, tree.GetDepth(left_child_nidx),
-                                    num_leaves)) {
-        monitor.StartCuda("UpdatePosition");
-        this->UpdatePosition(candidate.nid, (*p_tree)[candidate.nid]);
-        monitor.StopCuda("UpdatePosition");
+      if (ExpandEntry::ChildIsValid(param, tree.GetDepth(left_child_nidx), num_leaves)) {
+        for (auto& batch : p_fmat->GetBatches<EllpackPage>(batch_param)) {
```

Comment:
wouldn't iterating over *all* the batches *successively* while building root node, each node in the tree and finalizing the position for *every* iteration significantly add to the training time when external memory mode is enabled? it would be interesting to see the difference between this version and the one in master for one of the representative dataset when external memory is enabled.